// @cfxjs/react-hooks/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "useConfluxPortal", {
  enumerable: true,
  get: function () {
    return _useConfluxPortal.default;
  }
});
Object.defineProperty(exports, "useEpochNumber", {
  enumerable: true,
  get: function () {
    return _useEpochNumber.useEpochNumber;
  }
});
Object.defineProperty(exports, "useEpochNumberFn", {
  enumerable: true,
  get: function () {
    return _useEpochNumber.useEpochNumberFn;
  }
});
Object.defineProperty(exports, "useBalance", {
  enumerable: true,
  get: function () {
    return _useBalance.default;
  }
});
Object.defineProperty(exports, "useSWR", {
  enumerable: true,
  get: function () {
    return _swr.default;
  }
});
Object.defineProperty(exports, "useEpochNumberSWR", {
  enumerable: true,
  get: function () {
    return _swr.useEpochNumberSWR;
  }
});
Object.defineProperty(exports, "initContract", {
  enumerable: true,
  get: function () {
    return _initContract.default;
  }
});
Object.defineProperty(exports, "useConfluxJSDefined", {
  enumerable: true,
  get: function () {
    return _useConfluxJSDefined.default;
  }
});
Object.defineProperty(exports, "useStatus", {
  enumerable: true,
  get: function () {
    return _useStatus.default;
  }
});
Object.defineProperty(exports, "useConfirmationRiskByHash", {
  enumerable: true,
  get: function () {
    return _useConfirmationRiskByHash.default;
  }
});
Object.defineProperty(exports, "useClientVersion", {
  enumerable: true,
  get: function () {
    return _useClientVersion.default;
  }
});
Object.defineProperty(exports, "useChainNetId", {
  enumerable: true,
  get: function () {
    return _useChainNetId.default;
  }
});
Object.defineProperty(exports, "Big", {
  enumerable: true,
  get: function () {
    return _big.default;
  }
});

var _useConfluxPortal = _interopRequireDefault(require("./useConfluxPortal"));

var _useEpochNumber = require("./useEpochNumber");

var _useBalance = _interopRequireDefault(require("./useBalance"));

var _swr = _interopRequireWildcard(require("./swr"));

var _initContract = _interopRequireDefault(require("./initContract"));

var _useConfluxJSDefined = _interopRequireDefault(require("./useConfluxJSDefined"));

var _useStatus = _interopRequireDefault(require("./useStatus"));

var _useConfirmationRiskByHash = _interopRequireDefault(require("./useConfirmationRiskByHash"));

var _useClientVersion = _interopRequireDefault(require("./useClientVersion"));

var _useChainNetId = _interopRequireDefault(require("./useChainNetId.js"));

var _big = _interopRequireDefault(require("big.js"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _useEpochNumber.setupEpochListener)();
